### PR TITLE
perf(marshal): Extend new assert style to passStyleOf `check` calls

### PR DESCRIPTION
**Pull request by gibson042**
Ref dc24f2f2c3cac7ce239a64c503493c41a2334315

<details><summary>ad hoc analysis</summary>

```sh
$ node --input-type=module -e '
  import Benchmark from "benchmark";
  import "../init/index.js";
  import { passStyleOf } from "./src/passStyleOf.js";
  import { makeTagged } from "./src/makeTagged.js";
  const inputs = [
    undefined,
    "foo",
    true,
    33,
    33n,
    Symbol.for("foo"),
    Symbol.iterator,
    null,
    Promise.resolve(null),
    [3, 4],
    { foo: 3 },
    { then: "non-function then ok" },
    makeTagged("unknown", undefined),
    Error("ok"),
  ].map(v => harden(v));
  const badInputs = [
    Symbol("unique"),
    {},
    harden(Object.setPrototypeOf(Promise.resolve(), { __proto__: Promise.prototype })),
    harden(Object.assign(Promise.resolve(), { extra: "unexpected own property" })),
    harden(Object.defineProperties(Promise.resolve(), { then: { value: () => "bad then" } })),
    harden({ then: () => "thenable" }),
  ];
  let result;
  (new Benchmark.Suite())
    .add("valid inputs", () => {
      for (const v of inputs) result = passStyleOf(v);
    })
    .add("invalid inputs", () => {
      let caught;
      for (const v of badInputs) {
        caught = undefined;
        try {
          result = passStyleOf(v);
        } catch (err) {
          caught = err;
        } finally {
          if (!caught) throw new Error(`unexpected acceptance: ${v}`);
        }
      }
    })
    .add("big input", () => {
      result = passStyleOf(harden(Array(1000).fill().map(() => [...inputs])));
    })
    .on("cycle", evt => console.log(String(evt.target)))
    .on("complete", () => result)
    .run({ async: true })
'
```

</details>

before:
```
valid inputs x 831,564 ops/sec ±0.71% (91 runs sampled)
invalid inputs x 3,645 ops/sec ±0.86% (88 runs sampled)
big input x 10.31 ops/sec ±3.22% (32 runs sampled)
```
after:
```
valid inputs x 828,877 ops/sec ±0.60% (91 runs sampled)
invalid inputs x 3,780 ops/sec ±0.76% (85 runs sampled)
big input x 66.53 ops/sec ±1.74% (67 runs sampled)
```

(and we could probably get even more improvement by replacing `check` parameters with `reject` to avoid creating the latter functions inline)

### CODE REVIEW

**Review by erights — COMMENTED** [review]
Should I review this assuming that you intend it to be a pure refactoring --- with no observable differences? FWIW, I have not spotted any yet.

**Comment by gibson042** [comment]
> Should I review this assuming that you intend it to be a pure refactoring --- with no observable differences? FWIW, I have not spotted any yet.

Correct. :+1:

**Review by erights — APPROVED** [review]
All nice simplifications, thanks!

LGTM

File: packages/marshal/src/helpers/copyRecord.js
Comment (kriskowal) [inline packages/marshal/src/helpers/copyRecord.js:40]: @erights I’m reasonably certain that the X/describe tagged literal does not have the same effect on `reject` or even as an argument to the `Error` constructor as it does on `assert` methods. You may need `assert.error` here for the details to be extractable by the console.

**Review by kriskowal — COMMENTED** [review]
Looks mechanical to me, but please wait for confirmation about using details/quote (X/q) with `reject`. I’m pretty sure it doesn’t do what it says on the tin.

File: packages/marshal/src/helpers/copyRecord.js
Comment (gibson042) [reply packages/marshal/src/helpers/copyRecord.js:40]: `assertValid` calls `canBeValid(candidate, assertChecker)`, where `assertChecker` (and therefore `check` here) is `(cond, details) => { assert(cond, details); return true; }`, so this `reject` ultimately does send output from ``` X`…` ``` to `assert` in that case. Or are you saying that there's some dependence upon e.g. static analysis detecting literal use of "assert"?

Comment (kriskowal) [reply packages/marshal/src/helpers/copyRecord.js:40]: No, my review was too shallow. I see now you’re plumbing through `assert` and that `reject` is not `Promise.reject`.

**Review by kriskowal — APPROVED** [review]
Please dismiss my prior review.

**Pull request merged**
